pallets/inactivity-tracking/src/tests.rs

```diff
@@ -0,0 +1,573 @@
+        assert_eq!(
+            <Pallet::<Test> as NodeActivityTrackingHelper<AccountId>>::is_node_inactive(
+                &COLLATOR_1
+            ),
+            false
+        );
+        assert_eq!(
+            <Pallet::<Test> as NodeActivityTrackingHelper<AccountId>>::is_node_inactive(
+                &COLLATOR_2
+            ),
+            false
+        );
+        // Now start = CurrentSessionIndex so the collators should be considered active
+        roll_to(SESSION_BLOCK_LENGTH * (suspension_period as u64 + 2));
+        assert_eq!(
+            <Test as Config>::CurrentSessionIndex::session_index(),
+            suspension_period + 2
+        );
+        assert_eq!(
+            <Pallet::<Test> as NodeActivityTrackingHelper<AccountId>>::is_node_inactive(
+                &COLLATOR_1
+            ),
+            false
+        );
+        assert_eq!(
+            <Pallet::<Test> as NodeActivityTrackingHelper<AccountId>>::is_node_inactive(
+                &COLLATOR_2
+            ),
+            false
+        );
+        // Once CurrentSessionIndex >= start + MaxInactiveSessions collators will be considered inactive
+        // since there are inactivity records for it
+        roll_to(
+            SESSION_BLOCK_LENGTH
+                * (suspension_period as u64 + 2 + get_max_inactive_sessions() as u64),
+        );
+        assert_eq!(
+            <Test as Config>::CurrentSessionIndex::session_index(),
+            suspension_period + 2 + get_max_inactive_sessions()
+        );
+        assert_eq!(
+            <Pallet::<Test> as NodeActivityTrackingHelper<AccountId>>::is_node_inactive(
+                &COLLATOR_1
+            ),
+            true
+        );
+        assert_eq!(
+            <Pallet::<Test> as NodeActivityTrackingHelper<AccountId>>::is_node_inactive(
+                &COLLATOR_2
+            ),
+            true
+        );
+    });
+}
+
+#[test]
+fn processing_ended_session_correctly_updates_current_session_collators_and_active_collators_records(
+) {
+    ExtBuilder.build().execute_with(|| {
+        let current_session_active_collator_record: BoundedBTreeSet<AccountId, ConstU32<5>> =
+            get_collator_set(vec![COLLATOR_1]);
+        let inactive_collator_record: BoundedBTreeSet<AccountId, ConstU32<5>> =
+            get_collator_set(vec![COLLATOR_2]);
+        let empty_vec: BoundedBTreeSet<AccountId, ConstU32<5>> = BoundedBTreeSet::new();
+
+        ActiveCollatorsForCurrentSession::<Test>::put(
+            current_session_active_collator_record.clone(),
+        );
+
+        assert_eq!(
+            ActiveCollatorsForCurrentSession::<Test>::get(),
+            current_session_active_collator_record
+        );
+        assert_eq!(InactiveCollators::<Test>::get(0), empty_vec);
+
+        roll_to(SESSION_BLOCK_LENGTH - 1);
+        Pallet::<Test>::on_before_session_ending();
+        roll_to(SESSION_BLOCK_LENGTH);
+        Pallet::<Test>::process_ended_session();
+
```

Comment:
Having to call `on_before_session_ending` and `process_ended_session` manually in each test is very error prone. Why don't we integrate it into `roll_to`?

I see another pallet uses

```
impl pallet_session::SessionHandler<u64> for TestSessionHandler {
```

not sure what's the difference between that and what you have here, if you can use that it's probably better

Reply:
For sure. Added the setup in https://github.com/moondance-labs/tanssi/pull/922/commits/00b45e565658b359c373ebc659269eb8f88c8ecf